runtime/doc/api.txt

```diff
@@ -3358,13 +3358,27 @@ nvim_win_text_height({window}, {opts})                *nvim_win_text_height()*
                     0-based inclusive, rounded down to full screen lines. When
                     omitted include the whole line.
                   • end_vcol: Ending virtual column index on "end_row",
-                    0-based exclusive, rounded up to full screen lines. When
-                    omitted include the whole line.
+                    0-based exclusive, rounded up to full screen lines. When 0
+                    only include diff filler and virtual lines above
+                    "end_row". When omitted include the whole line.
+                  • max_height: Don't add the height of lines below the row
+                    for which this height is reached. Useful to e.g. limit the
+                    height to the window height, avoiding unnecessary work. Or
+                    to find out how many buffer lines beyond "start_row" take
+                    up a certain number of logical lines (returned in
+                    "end_row" and "end_vcol").
 
     Return: ~
         Dict containing text height information, with these keys:
         • all: The total number of screen lines occupied by the range.
         • fill: The number of diff filler or virtual lines among them.
+        • end_row: The row for which "max_height" is reached (first row of a
+          closed fold), or "end_row" from {opts} if "max_height" isn't
+          reached.
+        • end_vcol: Number of columns used up by "end_row" to reach
+          "max_height", or if "max_height" isn't reached, either "end_vcol"
+          from {opts} or the number of columns "end_row" takes up on the
```

Comment:
`end_vcol` is not the number of columns used up by `end_row` when `start_vcol` and `end_vcol` from `opts` are on the same line (e.g. with a very long line and `'smoothscroll'`), but I'm not sure how else to describe it.

Reply:
Yes, makes sense.